pvlib/atmosphere.py

```diff
@@ -422,16 +431,18 @@ def first_solar_spectral_correction(pw, airmass_absolute, module_type=None,
     # *** Pwat ***
     # Replace Pwat Values below 0.1 cm with 0.1 cm to prevent model from
     # diverging"
-
-    if np.min(pw) < 0.1:
-        pw = np.maximum(pw, 0.1)
-        warn('Exceptionally low Pwat values replaced with 0.1 cm to prevent' +
-             ' model divergence')
+    pw = np.atleast_1d(pw)
+    pw = pw.astype('float64')
```

Comment:
why is the type casting necessary?

Reply:
It is necessary for the case when the input is an integer (or array of integers). In this case, the data type of the array is int32 and is not compatible with np.nan (which is a float) for the executing of `pw[pw > max_pw] = np.nan`. It raises an error if I don't cast the type here.